### PR TITLE
feat(pat-4140): Add TimeField to C#; add missing boolean ops

### PR DESCRIPTION
**Pull request by ajmasci**
- Add `TimeField` class to static C#
- Add missing boolean ops to supported temporal field classes.

### Test plan
TBD

### CODE REVIEW

**Comment by ajmasci** [comment]
Current dependencies on/for this PR:
* main
  * **PR #143** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/143" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/patch-tech/dpm/143?utm_source=stack-comment).

File: static/csharp/Dpm/TimeField.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    public static BinaryBooleanFieldExpr operator >=(TimeField a, TimeOnly b)
+    {
+      return new BinaryBooleanFieldExpr(a, BooleanOperatorType.gte, new LiteralField<string>(b.ToString("O")));
+    }
+
+    /// <summary>
+    /// Returns a boolean expression that performs a relative range check of this TimeField.
+    /// The range is specified by its two bounds and a granularity.
+    /// E.g., the filter expression below checks if the value of `startDate` lies
+    /// in the past 2 to 3 hours.
```
Comment (spencerwilson) [inline static/csharp/Dpm/TimeField.cs:135]: Could you make this description more explicit about whether either of the endpoints are included?

File: static/csharp/Dpm/LiteralField.cs [hunk 2]
```diff
@@ -85,7 +85,7 @@ static Query.Types.Literal makeLit(T s)
         return lit;
       }
 
-      if (IsList())
+      if (Values != null)
```
Comment (spencerwilson) [inline static/csharp/Dpm/LiteralField.cs:88]: Could you describe the change to this line?

**Review by spencerwilson — APPROVED** [review]
Since we're in the office, we can chat about my comments in person

File: static/csharp/Dpm/LiteralField.cs (hunk 2 above)
Comment (ajmasci) [reply static/csharp/Dpm/LiteralField.cs:88]: The change makes the C# compiler warning that there is a potential null value being used on L91 go away, even though the `IsList` function does the same check.

File: static/csharp/Dpm/Table.cs [hunk 3]
```diff
@@ -171,11 +171,11 @@ public Table Select(params FieldExpr[] selection)
...
     /// var query = MyTable.select(
     ///   name,
-    ///   MyTable["CATEGORY"],
-    ///   saleDate.month.As("saleMonth"),
+    ///   category,
+    ///   saleDate.Month.As("saleMonth"),
     ///   price.Avg().As("meanPrice")
     /// )
-    /// .OrderBy((MyTable["meanPrice"], "DESC'), (saleDate.Month, "ASC"))
+    /// .OrderBy((price.Avg(), "DESC'), (saleDate.Month, "ASC"))
```
Comment (spencerwilson) [inline static/csharp/Dpm/Table.cs:178]: Maybe?
```suggestion
    /// .OrderBy((price.Avg(), "DESC'), (saleDate.Month(), "ASC"))
```

**Review by spencerwilson — APPROVED** [review]

File: static/csharp/Dpm/Table.cs (hunk 3 above)
Comment (ajmasci) [reply static/csharp/Dpm/Table.cs:178]: `.Month` is a getter so no `()` required.

**Pull request merged**